pkg/controllers/routing/network_routes_controller.go

```diff
@@ -1496,3 +1416,182 @@ func NewNetworkRoutingController(clientset kubernetes.Interface,
 
 	return &nrc, nil
 }
+
+type bgpPeerConfig struct {
```

Comment:
I've been slowly trying to pull functionality out of the monolith controllers. I would do that here. As a matter of fact, there is a relatively recent `bgp` package at the top `pkg` level now.

Probably the best thing to do would be to externalize this, struct and its related functions (although I think most of them already are) and then put them in that package to continue grouping BGP functionality together.

Reply:
Moved this into the `bgp` package. I also noticed that there's a lot of duplication with the validation logic for the BGP peer configs, so I decided to lift that out from the routing package as well and put it into the `bgp` package.